### ISSUE TITLE
test: check-dashboard

### ISSUE THREAD

**Issue by dperpeet** [cockpit-project/cockpit#3147]
There is a recurring test failure:

```
# ----------------------------------------------------------------------
# testEdit (check_dashboard.TestDashboard)
#
Wrote TestDashboard-testEdit-FAIL.png
Warning: Permanently added '10.111.119.167' (ECDSA) to the list of known hosts.

Journal database copied to TestDashboard-testEdit-10.111.119.167-FAIL.journal
Journal extracted to TestDashboard-testEdit-10.111.119.167-FAIL.log
not ok 61 testEdit (check_dashboard.TestDashboard)
Traceback (most recent call last):
  File "check-dashboard", line 175, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "/build/cockpit/test/testlib.py", line 235, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "/build/cockpit/test/testlib.py", line 211, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/build/cockpit/test/testlib.py", line 670, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "/build/cockpit/test/testlib.py", line 693, in _invoke
    raise Error(res['error'])
Error: timeout
```

http://files.cockpit-project.org/logs/pull-3124-7c0c23a3-fedora-23-x86_64/
https://fedorapeople.org/groups/cockpit/logs/pull-3143-5b267ee0-rhel-7-x86_64/

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 175, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "/build/cockpit/test/testlib.py", line 236, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "/build/cockpit/test/testlib.py", line 212, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/build/cockpit/test/testlib.py", line 674, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "/build/cockpit/test/testlib.py", line 697, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://fedorapeople.org/groups/cockpit/logs/pull-3150-74c35775-fedora-22-x86_64/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 175, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "/build/cockpit/test/testlib.py", line 236, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "/build/cockpit/test/testlib.py", line 212, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/build/cockpit/test/testlib.py", line 674, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "/build/cockpit/test/testlib.py", line 697, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://fedorapeople.org/groups/cockpit/logs/pull-3138-9fd74295-fedora-22-x86_64/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 175, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 236, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 212, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 674, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 697, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://files.cockpit-project.org/logs/pull-3161-ef95ed8e-fedora-23-x86_64/log.html

**Comment by mvollmer** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 175, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 236, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 212, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 674, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 697, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://fedorapeople.org/groups/cockpit/logs/pull-3138-482843b1-rhel-7-x86_64/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 175, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 235, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 211, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 673, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 696, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://fedorapeople.org/groups/cockpit/logs/pull-3179-e7952e29-fedora-22-x86_64/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 175, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 235, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 211, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 673, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 696, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://fedorapeople.org/groups/cockpit/logs/pull-3184-31ba0687-rhel-7-x86_64/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 175, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 235, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 211, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 673, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 696, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://files.cockpit-project.org/logs/pull-3188-1d1122e3-fedora-23-x86_64/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 217, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 235, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 211, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 678, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 701, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://fedorapeople.org/groups/cockpit/logs/pull-3201-778b566e-rhel-7/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 217, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 235, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 211, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 678, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 701, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://files.cockpit-project.org/logs/pull-3249-6031b2bb-fedora-23/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 217, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 235, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 211, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 678, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 701, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://fedorapeople.org/groups/cockpit/logs/pull-3261-6500af38-rhel-7/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 217, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 237, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 213, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 698, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 721, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://files.cockpit-project.org/logs/pull-3261-836424e7-fedora-23/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 217, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 235, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 211, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 678, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 701, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://fedorapeople.org/groups/cockpit/logs/pull-3274-891d14f1-rhel-7/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 217, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 242, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 218, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 685, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 708, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://fedorapeople.org/groups/cockpit/logs/pull-3251-06f28fb2-rhel-7/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 217, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 242, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 218, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 687, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 710, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://files.cockpit-project.org/logs/pull-3324-e280bf18-fedora-23/log.html

**Comment by cockpituous** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 217, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 242, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 218, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 687, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 710, in _invoke
    raise Error(res['error'])
Error: timeout
```
http://fedorapeople.org/groups/cockpit/logs/pull-3334-8836d7c1-rhel-7/log.html

**Comment by mvollmer** [comment]
Ooops, it happened again
```
Traceback (most recent call last):
  File "check-dashboard", line 218, in testEdit
    b.wait_not_attr("#dashboard-hosts .list-group-item[data-address='localhost']", "style", old_style)
  File "testlib.py", line 242, in wait_not_attr
    return self.wait_js_func('!ph_has_attr', selector, attr, val)
  File "testlib.py", line 218, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 687, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 710, in _invoke
    raise Error(res['error'])
Error: timeout
```

**Comment by petervo** [comment]
Possible fix pushed in #3366
